src/coreclr/jit/llvmlower.cpp

```diff
@@ -1128,14 +1111,10 @@ unsigned Llvm::lowerCallToShadowStack(GenTreeCall* callNode)
             if (!argNode->OperIs(GT_FIELD_LIST) && argNode->TypeIs(TYP_STRUCT))
             {
                 normalizeStructUse(argNode, _compiler->typGetObjLayout(clsHnd));
+                callArg->SetSignatureClassHandle(clsHnd);
```

Comment:
```suggestion
                // TODO-LLVM: delete (together with 'SetSignatureClassHandle') when merging https://github.com/dotnet/runtime/pull/69969 (May 31).
                callArg->SetSignatureClassHandle(clsHnd);
```
I realize it is currently necessary to reset the class handle because of issues with canonical structs, but it should be done for all `TYP_STRUCT` args (this condition excludes `FIELD_LIST`s).